AdyenActions/Components/QRCode/QRCodeActionComponent.swift

```diff
@@ -84,34 +88,42 @@ public final class QRCodeActionComponent: ActionComponent, Cancellable {
     /// - Parameter timeoutInterval: QR Code expiration timeout
     internal init(context: AdyenContext,
                   configuration: Configuration = .init(),
-                  pollingComponentBuilder: AnyPollingHandlerProvider,
+                  pollingComponentBuilder: AnyPollingHandlerProvider? = nil,
                   timeoutInterval: TimeInterval) {
         self.context = context
         self.configuration = configuration
         self.pollingComponentBuilder = pollingComponentBuilder
         self.expirationTimeout = timeoutInterval
-        
+
         updateExpiration(timeoutInterval)
     }
 
     /// Handles QR code action.
     ///
     /// - Parameter action: The QR code action.
     public func handle(_ action: QRCodeAction) {
-        pollingComponent = pollingComponentBuilder.handler(for: action.paymentMethodType)
+        qrCodeAction = action
+        pollingComponent = pollingComponentBuilder?.handler(for: action.paymentMethodType)
         pollingComponent?.delegate = self
         
-        assert(presentationDelegate != nil)
+        AdyenAssertion.assert(message: "presentationDelegate is nil", condition: presentationDelegate == nil)
         
-        presentationDelegate?.present(
-            component: PresentableComponentWrapper(
+        let viewController = createViewController(with: action)
+        setUpPresenterViewController(parentViewController: viewController)
+
+        if let presentationDelegate = presentationDelegate {
+            let presentableComponent = PresentableComponentWrapper(
                 component: self,
-                viewController: createViewController(with: action)
+                viewController: viewController
             )
-        )
+            presentationDelegate.present(component: presentableComponent)
+        } else {
+            AdyenAssertion.assertionFailure(
+                message: "PresentationDelegate is nil. Provide a presentation delegate to QRCodeActionComponent."
+            )
+        }
         
         startTimer()
```

Comment:
you can pass the action object as a parameter here instead of keeping it an instance property, what do you think? :-)

Reply:
yes.